### ISSUE TITLE
Proposal: Use of dotmap or equivalent to simplify usage of the deserialized object

### ISSUE THREAD

**Issue by aronchick** [marshmallow-code/marshmallow#1555]
Because Marshmallow provides a nice abstraction over the datamodel, it might be nice to go even further and use something like DotMap (https://github.com/drgrib/dotmap) to simplify using the objects. 

E.g. instead of 
```
user['created_at']
```

you could type:

```
user.created_at
```

This also provides the benefit of giving you a layer to prevent messing with the object (instead of with a visible dict data structure, where it feels like it's more open to interference).

**Comment by lafrech** [comment]
You can use a post_load method to load deserialized data into a dotmap.

The docs provide an example of post_load method loading into an object.

**Comment by sloria** [comment]
Yes, a `post_load` method is a good place to do this. If you want to use `DotMap` as the dict class for both deserialization and serialization, you can override `dict_class`, like so:

```python
from dotmap import DotMap
from marshmallow import Schema, fields


class MySchema(Schema):
    dict_class = DotMap

    foo = fields.Str()


schema = MySchema()
loaded = schema.load({"foo": "bar"})
print(loaded)
# DotMap(foo='bar')
dumped = schema.dump(loaded)
print(dumped)
# DotMap(foo='bar')
```

**Comment by lafrech** [comment]
Good point. No need for a `post_load` method, here. Just supply `dict_class`.

**Comment by aronchick** [comment]
Wow, what a great idea! I asked a slightly different question here - https://stackoverflow.com/questions/61092278/structuring-an-object-in-python-to-direct-access-to-an-attribute

Copied over - basically, what's the easiest way to access the sub-fields? Thank you so much!

-------------

I'm building an object in Python using Marshmallow schema - and I was hoping to make the experience more elegant for users.

So, if I have an object like the following (stripped down to simplify):

```py
from dotmap import DotMap

class User:
    def __init__(self, name: str, kids: dict):
        self.name = name
        self.kids = DotMap(kids)
```

Then let's say I did the following:

```py
steve = User('steve', {'sally': 8, 'harry': 6}
steve.sally = 9
steve.harry = 7
```

Basically, instead of that first '.' accessing attributes on User, it's redirected to the DotMap. Is this possible? I realize this would make things challenging, because it would mean that all attributes on User (e.g. name) could be hidden by having the same key in the DotMap.

**Comment by deckar01** [comment]
Check out `__getitem__`.

https://docs.python.org/3/reference/datamodel.html#object.__getitem__
